README.md

```diff
@@ -1,2 +1,263 @@
+
+For example:
+```js
+removeProps('a')({a: 1, b: 2}) // returns {b: 2}
+
+removeProps(['a', 'b'])({a: 1, b: 2, c: 3}) // returns {c: 3}
+```
+
+### `mapToObject()`
+
+```js
+mapToObject(
+  callback: (value: any, key: string | number) => [key: string, value: any]
+): Function
+```
+
+Returns a function which "maps" the supplied callback to every element in its argument and builds a single object as its return value. The callback should return a two-element `[key, value]` array in the style of [`fromPairs()`](https://lodash.com/docs/4.17.11#fromPairs)
+
+For example:
+```js
+mapToObject((val, key) => [`${key}${val}`, `${val}${key}`])({a: 1, b: 2}) // returns {a1: '1a', b2: '2b'}
+```
+
+### `interleave()`
+
+```js
+interleave(
+  callback: ({following: any}) => any
+): Function
+```
+
+Returns a function which returns a new array with the results of calling the supplied callback "interleaved" between the elements of the array argument. The callback receives a `following` option-style argument which is the following array element
+
+For example:
+```js
+interleave(() => 'and then')(['a', 'b', 'c']) // returns ['a', 'and then', 'b', 'and then', 'c']
+
+interleave(
+  ({following}) => `the next item will be ${following}`
+)(['a', 'b', 'c']) // returns ['a', 'the next item will be b', 'b', 'the next item will be c', 'c']
+```
+
+### `fmapWithKey()`
+
+```js
+fmapWithKey(
+  callback: (value: any, key: string | number) => any
+): Function
+```
+
+Returns a function which "maps" the supplied callback to every element in its argument. The callback gets passed the value and key of the element
+
+For example:
+```js
+fmapWithKey((value, key) => `${key}${value}`)({a: 1, b: 2}) // returns ['a1', 'b2']
+```
+
+### `fmapValuesWithKey()`
+
+```js
+fmapValuesWithKey(
+  callback: (value: any, key: string | number) => any
+): Function
+```
+
+Returns a function which "maps" the supplied callback to every element in its argument and returns an object whose keys match that argument and whose values are the return values of the callback (like [`mapValues()`](https://lodash.com/docs/4.17.11#mapValues)). The callback gets passed the value and key of the element
+
+For example:
+```js
+fmapValuesWithKey((value, key) => `${key}${value}`)({
+  a: 1,
+  b: 2,
+}) // returns {a: 'a1', b: 'b2'}
+```
+
+### `feachWithKey()`
+
+```js
+feachWithKey(
+  callback: (value: any, key: string | number) => any
+): Function
+```
+
+Returns a function which applies the supplied callback to every element in its argument. The callback gets passed the value and key of the element
+
+For example:
+```js
+const array = []
+feachWithKey((value, key) => array.push(`${key}${value}`))({a: 1, b: 2})
+// now array is ['a1', 'b2']
+
+const array2 = []
+feachWithKey((value, key) => array2.push(`${key}${value}`))(['a', 'b'])
+// now array2 is ['0a', '1b']
+```
+
+### `fflatMapWithKey()`
+
+```js
+fflatMapWithKey(
```

Comment:
Same comment here about naming. What I'd consider a "real" `flatMap` would not output a different container type than what was inputted. Maybe `reduceAndFlatten`?

Reply:
Again, this is implemented in terms of `lodash/fp` `flatMap()`, which will always return an array but accepts either an object or an array (what its docs calls a "collection")

Reply:
👍